app/controllers/users_controller.rb

```diff
@@ -496,28 +498,61 @@ def is_local_username
     usernames -= groups
     usernames.each(&:downcase!)
 
-    cannot_see = []
+    users = User
+      .where(staged: false)
+      .where(username_lower: usernames)
+      .index_by(&:username_lower)
+
+    cannot_see = {}
     here_count = nil
 
     topic_id = params[:topic_id]
     if topic_id.present? && topic = Topic.find_by(id: topic_id)
+      topic_muted_by = TopicUser
+        .where(topic: topic)
+        .where(user_id: users.values.map(&:id))
+        .where(notification_level: TopicUser.notification_levels[:muted])
+        .pluck(:user_id)
+        .to_set
+
+      topic_allowed_user_ids = TopicAllowedUser
+        .where(topic: topic)
+        .where(user_id: users.values.map(&:id))
+        .pluck(:user_id)
+        .to_set
+
+      topic_allowed_group_ids = TopicAllowedGroup
+        .where(topic: topic)
+        .pluck(:group_id)
+        .to_set
```

Comment:
I believe these queries are only useful if the topic is a private message so we do not need to run the queries if for a normal topic.

Reply:
I added this under a `if topic.private_message?`.